tensorbay/opendataset/HKD/loader.py

```diff
@@ -0,0 +1,100 @@
+import csv
+import os
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.geometry import Keypoint2D
+from tensorbay.label import LabeledKeypoints2D
+from tensorbay.utility import chunked
+
+DATASET_NAME = "HKD"
+_CSV_NAMES = {
+    1: "subject1_fingercount_2D_Annotations_cropped",
+    2: "subject2_fingercount_2D_Annotations_cropped",
+    3: "fingerappose_subject3_2D_Annotations_cropped",
+    4: "subject4_2D_Annotations_cropped",
+}
+_IMAGE_NAMES = {
+    1: "subject1_fingercount_cropframe_",
+    2: "subject2_fingercount_cropframe_",
+    3: "fingerappose_subject3_cropframe_",
+    4: "subject4_cropframe_",
+}
+
+
+def HKD(path: str) -> Dataset:
+    """`HKD <http://vlm1.uta.edu/~srujana/HandPoseDataset/HKD_Cropped/>`_ dataset.
+
+    The file structure should be like:
+
+        <path>
+            AnnotatedData_subject1/
+                CropImages/
+                    subject1_fingercount_cropframe_2.jpg
+                    subject1_fingercount_cropframe_3.jpg
+                    ...
+                    subject1_fingercount_cropframe_210.jpg
+                subject1_fingercount_2D_Annotations_cropped.csv
+
+            AnnotatedData_subject2/
+                CropImages/
+                    subject2_fingercount_cropframe_2.jpg
+                    subject2_fingercount_cropframe_3.jpg
+                    ...
+                    subject2_fingercount_cropframe_207.jpg
+                subject2_fingercount_2D_Annotations_cropped.csv
+
+            AnnotatedData_subject3/
+                CropImages/
+                    fingerappose_subject3_cropframe_2.jpg
+                    fingerappose_subject3_cropframe_3.jpg
+                    ...
+                    fingerappose_subject3_cropframe_235.jpg
+                fingerappose_subject3_2D_Annotations_cropped.csv
+
+            AnnotatedData_subject4/
+                CropImages/
+                    subject4_cropframe_2.jpg
+                    subject4_cropframe_3.jpg
+                    ...
+                    subject4_cropframe_147.jpg
+                subject4_2D_Annotations_cropped.csv
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    for subject_id in range(1, 5):
+        segment = dataset.create_segment(segment_name="subject" + str(subject_id))
+        csv_path = os.path.join(
+            root_path, "AnnotatedData_subject" + str(subject_id), _CSV_NAMES[subject_id] + ".csv"
+        )
+        with open(csv_path, encoding="utf-8") as fp:
+            csv_lines = csv.reader(fp)
+            for csv_line in csv_lines:
+                image_path = os.path.join(
+                    root_path,
+                    f"AnnotatedDara_subject{subject_id}/CropImages/\
```

Comment:
the file name separator `/` is different on Windows system, use `os.path.join` to handle the separator.